### PR TITLE
lib: hijack: use journalctl command to dump kernel information

### PR DESCRIPTION
**Pull request by Bin-QA**
use journalctl instead of catch information from /var/log/kern.log file
1. DMESG_LOG_START_LINE => CASE_KERNEL_START_TIME
2. with this change, sof-get-kernel-line is not useful, remove it

Signed-off-by: Wu, BinX <binx.wu@intel.com>

### CODE REVIEW

File: case-lib/hijack.sh [hunk 1]
```diff
@@ -20,15 +20,10 @@ function func_exit_handler()
...
     fi
     # when case ends, store kernel log
-    # /var/log/kern.log format:
-    # f1    f2  f3   f4          f5      f6 f7    f8...
-    # Mouth day Time MachineName kernel: [  time] content
-    # May 15 21:28:38 MachineName kernel: [    6.469255] sof-audio-pci 0000:00:0e.0: ipc rx: 0x90020000: GLB_TRACE_MSG
-    # May 15 21:28:38 MachineName kernel: [    6.469268] sof-audio-pci 0000:00:0e.0: ipc rx done: 0x90020000: GLB_TRACE_MSG
-    if [[ -n "$DMESG_LOG_START_LINE" && "$DMESG_LOG_START_LINE" -ne 0 ]]; then
-        tail -n +"$DMESG_LOG_START_LINE" /var/log/kern.log |cut -f5- -d ' ' > "$LOG_ROOT/dmesg.txt"
+    if [[ -n "$CASE_KERNEL_START_TIME" ]]; then
```
Comment (marc-hb) [inline case-lib/hijack.sh:23]: Single `[ ]` are enough for this.

File: case-lib/lib.sh [hunk 2]
```diff
@@ -191,9 +191,8 @@ func_lib_get_tplg_path()
     return 0
 }
 
-# force ask buffer data write into file system
-sudo sync -f
-# catch kern.log last line as current case start line
-if [ ! "$DMESG_LOG_START_LINE" ]; then
-    DMESG_LOG_START_LINE=$(wc -l /var/log/kern.log|awk '{print $1;}')
+if [ ! "$CASE_KERNEL_START_TIME" ]; then
```
Comment (marc-hb) [inline case-lib/lib.sh:194]: I'm not a fan of the empty expression `[  ]`, do you mind the more explicit `[ -n "$CASE_KERNEL_START_TIME"  ]`?  Consistent with hijack.sh above

File: test-case/check-reboot.sh [hunk 3]
```diff
@@ -144,14 +144,14 @@ sudo chown $UID $boot_file
...
-# DMESG_LOG_START_LINE to just keep last kernel bootup log
+# CASE_KERNEL_START_TIME to just keep last kernel bootup log
 boot_once_flag=$(realpath $(which sof-boot-once.sh))
 cat << END > $boot_file
 $old_content
 
 $boot_once_flag
 export LOG_ROOT='$(realpath $LOG_ROOT)'
-export DMESG_LOG_START_LINE=$(wc -l /var/log/kern.log|awk '{print $1;}')
+export CASE_KERNEL_START_TIME='$(journalctl --dmesg --no-pager -n 1 |head -n 1|awk '{print $(NF-3)" "$(NF-2);}')'
```
Comment (marc-hb) [inline test-case/check-reboot.sh:154]: That doesn't seem to be the kernel start time, did you test this?

Anything wrong with `uptime -s`?

**Comment by xiulipan** [comment]
I will close this one as @Bin-QA no longer work on this.
@aiChaoSONG @marc-hb do we still need some similar issues? I will leave all journalctl related issue to you @marc-hb

**Comment by xiulipan** [comment]
Close with new attempt #468

**Pull request closed without merging**